Source/ASRangeManagedNode.h

```diff
@@ -0,0 +1,35 @@
+//
+//  ASRangeManagedNode.h
+//  Texture
+//
+//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#import <Foundation/Foundation.h>
+#import <AsyncDisplayKit/ASTraitCollection.h>
+
+@class ASCellNode;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Basically ASTableNode or ASCollectionNode.
+ */
+@protocol ASRangeManagedNode <NSObject, ASTraitEnvironment>
```

Comment:
Maybe I'm being overly semantic, but the methods these adopt don't seem to match the name. I can imagine nodes that range manage other nodes that don't have index paths. Anyway, wondering if there's a better name but I don't have suggestions.

Also, should it be `ASRangeManagingNode` instead? That would better align with `ASHierarchyStateRangeManaged` state naming.